integrationTests/chainSimulator/mempool/testutils_test.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/process/block/preprocess"
+	"github.com/multiversx/mx-chain-go/storage/txcache"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	oneEGLD                   = big.NewInt(1000000000000000000)
+	oneQuarterOfEGLD          = big.NewInt(250000000000000000)
+	durationWaitAfterSendMany = 750 * time.Millisecond
+	durationWaitAfterSendSome = 10 * time.Millisecond
+)
+
+func startChainSimulator(t *testing.T, alterConfigsFunction func(cfg *config.Configs)) testsChainSimulator.ChainSimulator {
+	simulator, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck: true,
+		TempDir:                t.TempDir(),
+		PathToInitialConfig:    "../../../cmd/node/config/",
+		NumOfShards:            1,
+		GenesisTimestamp:       time.Now().Unix(),
+		RoundDurationInMillis:  uint64(4000),
+		RoundsPerEpoch: core.OptionalUint64{
+			HasValue: true,
+			Value:    10,
+		},
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         1,
+		MetaChainMinNodes:        1,
+		NumNodesWaitingListMeta:  0,
+		NumNodesWaitingListShard: 0,
+		AlterConfigsFunction:     alterConfigsFunction,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, simulator)
+
+	err = simulator.GenerateBlocksUntilEpochIsReached(1)
+	require.NoError(t, err)
+
+	return simulator
+}
+
+type participantsHolder struct {
+	sendersByShard  map[int][]dtos.WalletAddress
+	relayerByShard  map[int]dtos.WalletAddress
+	receiverByShard map[int]dtos.WalletAddress
+}
+
+func newParticipantsHolder() *participantsHolder {
+	return &participantsHolder{
+		sendersByShard:  make(map[int][]dtos.WalletAddress),
+		relayerByShard:  make(map[int]dtos.WalletAddress),
+		receiverByShard: make(map[int]dtos.WalletAddress),
+	}
+}
+
+func createParticipants(t *testing.T, simulator testsChainSimulator.ChainSimulator, numSendersPerShard int) *participantsHolder {
+	numShards := int(simulator.GetNodeHandler(0).GetShardCoordinator().NumberOfShards())
+	participants := newParticipantsHolder()
+
+	for shard := 0; shard < numShards; shard++ {
+		senders := make([]dtos.WalletAddress, 0, numSendersPerShard)
+
+		for i := 0; i < numSendersPerShard; i++ {
+			sender, err := simulator.GenerateAndMintWalletAddress(uint32(shard), oneEGLD)
+			require.NoError(t, err)
+
+			senders = append(senders, sender)
+		}
+
+		relayer, err := simulator.GenerateAndMintWalletAddress(uint32(shard), oneEGLD)
+		require.NoError(t, err)
+
+		receiver, err := simulator.GenerateAndMintWalletAddress(uint32(shard), big.NewInt(0))
+		require.NoError(t, err)
+
+		participants.sendersByShard[shard] = senders
+		participants.relayerByShard[shard] = relayer
+		participants.receiverByShard[shard] = receiver
+	}
+
+	err := simulator.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	return participants
+}
+
+type noncesTracker struct {
```

Comment:
I think we should export these in further PRs and reuse them in other tests :+1:

Reply:
Added MX-16448 (a bit of effort to make them nicer & generic).